addons/mail/static/src/components/notification_list/notification_list.js

```diff
@@ -111,7 +111,7 @@ class NotificationList extends Component {
                 if (t1.localMessageUnreadCounter === 0 && t2.localMessageUnreadCounter > 0) {
                     return 1;
                 }
-                if (t1.lastMessage && t2.lastMessage) {
+                if (t1.lastMessage && t1.lastMessage.date && t2.lastMessage && t2.lastMessage.date) {
```

Comment:
This condition should probably be completely removed. It's not consistent with how we display messages to use the date for sorting purposes. We never trust the date because from email it is user input.

This should also allow to remove this "+2s" in tests.